modules/rostests/apitests/advapi32/RegCreateKeyExW.c

```diff
@@ -0,0 +1,158 @@
+
+// Based on code from the following:
+// https://learn.microsoft.com/en-us/windows/win32/secauthz/creating-a-security-descriptor-for-a-new-object-in-c--
+
+#include <apitest.h>
+#include <stdio.h>
+
+#define WIN32_NO_STATUS
+#define _INC_WINDOWS
+#define COM_NO_WINDOWS_H
+#include <windef.h>
+#include <aclapi.h>
+#include <tchar.h>
+#include <debug.h>
+
+
+START_TEST(RegCreateKeyExW)
+{
+    HKEY hkey_main;
+    DWORD dwRes, dwDisposition;
+    PACL pACL = NULL;
+    PSECURITY_DESCRIPTOR pSD = NULL;
+    PSID pEveryoneSID = NULL, pAdminSID = NULL;
+    SID_IDENTIFIER_AUTHORITY SIDAuthWorld = {SECURITY_WORLD_SID_AUTHORITY};
+    SID_IDENTIFIER_AUTHORITY SIDAuthNT = {SECURITY_NT_AUTHORITY};
+    EXPLICIT_ACCESS ea[2];
+    SECURITY_ATTRIBUTES sa;
+    LONG lRes;
+    BOOL bRes;
+    LONG ErrorCode = 0;
+    HKEY hkSub = NULL;
+
+    /* If any of the test keys already exist, delete them to ensure proper testing*/
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey", KEY_ALL_ACCESS, 0);
```

Comment:
You need to close `hkey_main` before deleting.

Reply:
My understanding is that if I am using a 'predefined key' (HKEY_CURRENT_USER here for instance), then there is no need to close the key. Just for good measure and to be consistent I will add the close to this code. Thanks.

Reply:
The "mykey" test subkey is not a predefined key, so you need to close that.

Reply:
Added close code. Thanks.

Reply:
> direct copy of what was used in 'RegQueryValueExW.c" test, so maybe it needs some attention as well.

I don't see what would be wrong there: especially, that code **does** close its keys.